pkg/controllers/vdb/init_db.go

```diff
@@ -365,45 +332,26 @@ func (g *GenericDatabaseInitializer) getAzureAuthParmsContent(ctx context.Contex
 	}
 	azureConfigJSON.WriteString("}]")
 
-	content := fmt.Sprintf(`
-	  AzureStorageCredentials = %s
-	  AzureStorageEndpointConfig = %s
-	  %s
-	`, azureCredsJSON.String(), azureConfigJSON.String(), g.getCAFile())
-	return dedent.Dedent(content), ctrl.Result{}, nil
+	g.ConfigurationParams.Set("AzureStorageCredentials", azureCredsJSON.String())
+	g.ConfigurationParams.Set("AzureStorageEndpointConfig", azureConfigJSON.String())
+	g.setCAFile()
+	return ctrl.Result{}, nil
 }
 
-// getAdditionalConfigParmsContent constructs a string containing additional server config parameters
-func (g *GenericDatabaseInitializer) getAdditionalConfigParmsContent(content string) string {
-	parmNames := genCommunalParmsMap(content)
-	parms := strings.Builder{}
+// setAdditionalConfigParms adds additional server config parameters
+// to the config parms map.
+func (g *GenericDatabaseInitializer) setAdditionalConfigParms() {
 	for k, v := range g.Vdb.Spec.Communal.AdditionalConfig {
 		// we lowercase parm names to catch duplications
 		// like AWSauth/awsauth. In case of duplicate names,
 		// we skip to the next parm.
```

Comment:
I think we can remove this comment now as the lowercase logic was moved to the CiMap